libs/extended/src/System/Logger/Extended.hs

```diff
@@ -1,32 +1,90 @@
---    This will add a dependency on aeson for this package,
---    but it already transitively depended on it through imports.
---    Interestingly, the only place where imports uses Aeson
---    is in the Orphans module which defines Aeson Orphans instances for Log.Level.
---    So while we're at it, we should probably move those orphans here.
-mkLogger :: Log.Level -> Bool -> IO Log.Logger
-mkLogger lvl netstr = Log.new
+
+deriving instance Generic LC.Level
+instance FromJSON LC.Level
+instance ToJSON LC.Level
+
+-- | The log formats supported
+data LogFormat = JSON | Plain | Netstring 
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (ToJSON, FromJSON) -- TODO write this instance manually?
+
+-- | We use this as an intermediate structure to ease the implementation of the
+-- ToJSON instance but we could just inline everything. I think this has
+-- negligible impact and makes the code a bit more readable. Let me know
+data Element' = Element' [(Builder, Builder)] [Builder]
+
+instance ToJSON Element' where
+  toJSON = error "Use toEncoding"
+  toEncoding :: Element' -> Encoding
+  toEncoding (Element' fields msgs) = pairs $ fieldsToSeries fields <> msgsToSeries msgs
+    where
+      fieldsToSeries :: [(Builder, Builder)] -> Series
+      fieldsToSeries = foldMap $ \(k,v) -> pair (cs . eval $ k) (text . cs . eval $ v)
+      msgsToSeries :: [Builder] -> Series
+      msgsToSeries  = pair "msgs" . list (text . cs . eval)
+
+collect :: [Element] -> Element'
+collect = foldr go (Element' [] []) 
+  where
+    go :: Element -> Element' -> Element'
+    go (Bytes b) (Element' f m) =
+      Element' f (b : m)
+    go (Field k v) (Element' f m) = 
+      Element' ((k,v) : f) m
+
+jsonRenderer :: Renderer
+jsonRenderer _sep _dateFormat _logLevel = fromEncoding  . toEncoding . collect
+
+-- | Here for backwards-compatibility reasons
+netStringsToLogFormat :: Bool -> LogFormat
+netStringsToLogFormat True = Netstring
+netStringsToLogFormat False = Plain
+
+-- | Creates a logger given a log format Also takes an useNetstrings argument
+-- which is there because we cannot immediatelly deprecate the old interface.
+-- Old configs only provide the useNetstrings argument and not the logFormat
+-- argument, and in that case implement the old behaviour of either enabling
+-- plain text logging or netstring logging.  If both arguments are set,
+-- logFormat takes presedence over useNetstrings
+--
+-- TODO: Once we get rid of the useNetstrings in our config files, we can
+-- change the type of mkLogger to  mkLogger :: Log.Level -> LogFormat -> IO
+-- Log.Logger
+mkLogger :: Log.Level -> Maybe (Last Bool) -> Maybe (Last LogFormat) -> IO  Log.Logger
+mkLogger lvl useNetstrings logFormat = do
+  mkLogger'' lvl $
+    case (fmap netStringsToLogFormat <$> useNetstrings) <> logFormat of
+      Just x -> getLast x
+      Nothing -> Plain
+    
+mkLogger'' :: Log.Level -> LogFormat -> IO Log.Logger
```

Comment:
Call this `mkLoggerNew`, add a comment that it's a version of `mkLogger` that does not support the deprecated `useNetstrings` flag; and mention it in the haddocs of `mkLogger` instead of repeading the simplified type signature.